src/operation.js

```diff
@@ -557,7 +579,14 @@ export class Operation {
 
                 if (attrs.signer()) {
                     let signer = {};
-                    signer.pubKey = accountIdtoAddress(attrs.signer().pubKey());
+                    if (attrs.signer().key()._arm == "ed25519") {
```

Comment:
Thx, I couldn't find it. Would be great to have a small API reference in js-xdr. I will try to update union [example file](https://github.com/stellar/js-xdr/blob/master/examples/union.js).